### PR TITLE
Prepare for golangci-lint 1.50.1

### PR DESCRIPTION
**Pull request by skitt**
See individual commits for details.

### CODE REVIEW

File: pkg/syncer/test/util.go [hunk 1]
```diff
@@ -44,36 +44,38 @@ const (
...
 	meta, err := metaapi.Accessor(obj)
 	Expect(err).To(Succeed())
 
-	return resourceInterface.Get(context.TODO(), meta.GetName(), metav1.GetOptions{})
+	return resourceInterface.Get(ctx, meta.GetName(), metav1.GetOptions{})
 }
 
-func GetResource(resourceInterface dynamic.ResourceInterface, obj runtime.Object) *unstructured.Unstructured {
-	resource, err := GetResourceAndError(resourceInterface, obj)
+func GetResource(ctx context.Context, resourceInterface dynamic.ResourceInterface, obj runtime.Object) *unstructured.Unstructured {
```
Comment (tpantelis) [inline pkg/syncer/test/util.go]: Why is this necessary now?

Comment (skitt) [reply pkg/syncer/test/util.go]: `contextcheck` complains about it; we could disable that instead.

Comment (tpantelis) [reply pkg/syncer/test/util.go]: Disable it for test code.

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]
